app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java

```diff
@@ -179,4 +197,149 @@ public static JSONObject getContinents(Activity activity){
+     * @param tags
+     * @return
+     */
+    public static Post createPost(Activity activity, final Context context, String description, String title, ParseFile imageFile, ParseUser user, LatLng point, ArrayList<String> tags) {
+        final Post newPost = new Post();
+        newPost.setDescription(description);
+        if (imageFile == null) {
+            newPost.remove(com.example.footprnt.Util.Constants.image);
+        } else {
+            newPost.setImage(imageFile);
+        }
+        newPost.setUser(user);
+        newPost.setTitle(title);
+        newPost.setLocation(new ParseGeoPoint(point.latitude, point.longitude));
+        Geocoder gcd = new Geocoder(context, Locale.getDefault());
+        List<Address> addresses = null;
+        try {
+            addresses = gcd.getFromLocation(point.latitude, point.longitude, 1);
+            if (addresses.size() > 0) {
+                String city = addresses.get(0).getLocality();
+                String country = addresses.get(0).getCountryName();
+                String country_code = addresses.get(0).getCountryCode();
+                if (city != null) {
+                    newPost.setCity(city);
+                }
+                if (country != null) {
+                    newPost.setCountry(country);
+                }
+                JSONObject continents = getContinents(activity);
+                if (country_code != null && continents.has(country_code)) {
+                    String continent = continents.getString(country_code);
+                    newPost.setContinent(continent);
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+        newPost.setTags(tags);
+        newPost.saveInBackground(new SaveCallback() {
+            @Override
+            public void done(com.parse.ParseException e) {
+                if (e == null) {
+                    Toast.makeText(context, R.string.post_message, Toast.LENGTH_SHORT).show();
+                } else {
+                    e.printStackTrace();
+                }
+            }
+        });
+        return newPost;
+    }
+
+    /**
+     * Create a Google Map marker at specified point with given marker details
+     *
+     * @param md marker detail
+     */
+    public static Marker createMarker(GoogleMap mMap, MarkerDetails md) throws com.parse.ParseException {
+        Post post = (Post) md.getPost();
+        double latitude = (post.fetchIfNeeded().getParseGeoPoint("location")).getLatitude();
+        double longitude = (post.fetchIfNeeded().getParseGeoPoint("location")).getLongitude();
+        String title = (post.fetchIfNeeded().getString("title"));
+        ParseFile image = post.fetchIfNeeded().getParseFile("image");
+        String imageUrl = "";
+        if (image != null) {
+            imageUrl = image.getUrl();
+        }
+        Marker m = mMap.addMarker(new MarkerOptions()
```

Comment:
This is another case of you doing a good job of realizing a good piece of functionality that can be separated but not fully separating it from the rest of the code. I'd really like to see an API more like
  public static MarkerOptions createMarker(MarkerDetails md) throws ParseException {
when you do
  mMap.addMarker()
in whatever code calls this method.

Also please import `com.parse.ParseException` so that you can just use ParseException